test cases/common/33 run program/check-mingw.py

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env python3
+
+import os, sys, re
+
+if 'MSYSTEM' in os.environ and os.environ['MSYSTEM'] != '':
+    print(os.environ['MSYSTEM'])
+else:
+    match = re.search('[\\\\/](mingw32|mingw64|clang32|clang64|clangarm64|ucrt64)[\\\\/]', sys.executable, flags=re.IGNORECASE)
```

Comment:
I would suggest using `r'string'` here to avoid backslash escaping issues.

Reply:
Done at 9165d0b